org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/handler/StackTraceRequestHandler.java

```diff
@@ -64,8 +64,8 @@ public void handle(Command command, Arguments arguments, Response response, IDeb
                         ? thread.frames(stacktraceArgs.startFrame, totalFrames - stacktraceArgs.startFrame)
                         : thread.frames(stacktraceArgs.startFrame,
                         Math.min(totalFrames - stacktraceArgs.startFrame, stacktraceArgs.levels));
-                for (int i = 0; i < stacktraceArgs.levels; i++) {
-                    StackFrame stackFrame = stackFrames.get(stacktraceArgs.startFrame + i);
+                for (int i = 0; i < stackFrames.size(); i++) {
+                    StackFrame stackFrame = stackFrames.get(i);
                     int frameId = context.getRecyclableIdPool().addObject(stackFrame.thread().uniqueID(),

```

Comment:
remove the IndexOutOfBoundsException?

Reply:
no, when level is 0, this code always returns a empty array.